vagrant/Vagrantfile

```diff
@@ -1,32 +1,62 @@
+        setupControlplaneVip: true,               # this action is performed only if there are 2 or more machines with Master role  (see ClusterAPI, MachineSets/Role Master)
+        kubeadmInit: false, 
+        applyNetwork: false,
+        kubeadmJoinMaster: false,                 # this action is performed only if there are 2 or more machines with Master role  (see ClusterAPI, MachineSets/Role Master)
+        kubeadmJoin: false                        # this action is performed only if machines with Node role are defined            (see ClusterAPI, MachineSets/Role Node)
+    )
+    #.injectKubeadmFromBazelBuild(prefix: nil)    # uncomment if you want to inject a kubeadm release from bazel build; set the prefix if you want to differentiate the binary name
+    #.injectKubeadmFromDockerBuild(prefix: nil)   # uncomment if you want to inject a kubeadm release from docker build; set the prefix if you want to differentiate the binary name
+    #.injectKubeadmFromLocalBuild(prefix: nil)    # uncomment if you want to inject a kubeadm release from local build; set the prefix if you want to differentiate the binary name
+
+END {
+  TestGuide.print(clusterSpec)  
+}
 
-  config.vm.provider "virtualbox" do |vb|
-    vb.cpus   = "4"
-    vb.memory = "4096"
+Vagrant.configure("2") do |config|
+  if Vagrant.has_plugin?("vagrant-vbguest")
+    config.vbguest.auto_update = false
   end
 
-  config.vm.provision "shell", inline: <<-SHELL
-    #!/bin/sh
-    set -ex
-    
-    apt-get update && apt-get install -y apt-transport-https
-    curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
-    echo "deb http://apt.kubernetes.io/ kubernetes-xenial main" > /etc/apt/sources.list.d/kubernetes.list
-    apt-get update && apt-get install -y \
-      docker.io=1.13.1-0ubuntu6 \
-      kubelet=1.9.7-00 \
-      kubectl=1.9.7-00 \
-      kubeadm=1.9.7-00
-    
-    swapoff -a
-    systemctl enable docker.service
+  clusterSpec.machines.each do |machineSpec|
+    config.vm.define "#{machineSpec.name}" do |machine|
+      machine.vm.box = "#{machineSpec.box}"
+      machine.vm.hostname = "#{machineSpec.hostname}"
+      machine.vm.network :private_network,
+        :ip => "#{machineSpec.ip}",
+        :libvirt__forward_mode => "none",
+        :libvirt__network_name => "kubernetes",
+        :libvirt__netmask => "255.255.255.0",
+        :virtualbox__intnet => "kubenetwork"
+      machine.vm.provider "virtualbox" do |v|
+        v.name = "#{machineSpec.name}"
+  	    v.memory = "#{machineSpec.memory}"
+        v.cpus = "#{machineSpec.cpus}"
+        v.customize ["modifyvm", :id, "--uartmode1", "disconnected"]
+        v.linked_clone = true if Vagrant::VERSION =~ /^1.8/
+      end
+
+      if machineSpec.name == clusterSpec.machines.last.name
+        machine.vm.provision :ansible do |ansible|
+          ansible.compatibility_mode = "2.0"
```

Comment:
This looks like a neat option -- does it allow us to care less about what ansible version we depend on?

Reply:
Yes, it avoids some ansible warning and force you tu use latest formats for inventory file and so on